types/src/proof/definition.rs

```diff
@@ -602,10 +602,57 @@ impl SparseMerkleRangeProof {
         Self { right_siblings }
     }
 
-    /// Returns the siblings.
+    /// Returns the right siblings.
     pub fn right_siblings(&self) -> &[HashValue] {
         &self.right_siblings
     }
+
+    /// Verifies that the rightmost known leaf exists in the tree and that the resulting
+    /// root hash matches the expected root hash.
+    pub fn verify(
+        &self,
+        expected_root_hash: HashValue,
+        left_siblings: Vec<HashValue>,
+        rightmost_known_leaf: SparseMerkleLeafNode,
+    ) -> Result<()> {
+        let num_siblings = left_siblings.len() + self.right_siblings.len();
+        let mut left_sibling_iter = left_siblings.iter().rev();
```

Comment:
Now that it's a separate method, requiring the left / right siblings to be in the same order makes more sense.  If we really care about the cost, document it.

Reply:
On a side note: The `RangeProof` is really `PartialTreeProof`, right? The original name make me think it proof a chunk of leafs, but it's actually irrelevant with the actual chunk being received, instead, it always prove the entire left part of the tree, to a certain leaf. @wqfish

Reply:
1. Regarding, ordering, sounds good! I'll update the code (I didn't have a preference, just moved things across 😄)
2. Yes, exactly, that's my understanding, too. It's not really the range proof. We could rename it, but I won't do that here/now given the implications.

Reply:
1. taking an `impl Iterator<Item = HashValue>` won't be too ugly and avoids the `reverse()` IMO, now sure how you feel about it.

Reply:
Yeah, I'll leave it for now. The siblings shouldn't be huge, so we should be fine for now. If this becomes a problem in the future, we can deal with it 😄